test/web-platform-tests/run-wpts.js

```diff
@@ -61,29 +63,28 @@ describe("web-platform-tests", () => {
           });
 
           const testFile = testFilePath.slice((toRunDoc.DIR + "/").length);
-          let reason;
+          let reason, data;
 
           if (matchingPattern) {
             reason = toRunDoc[matchingPattern][0];
+            if (!Array.isArray(toRunDoc[matchingPattern])) {
+              data = toRunDoc[matchingPattern];
+            }
           } else if (toRunDoc.DIR.startsWith("html/canvas/")) {
             reason = "needs-canvas";
           }
 
-          const shouldSkip = ["fail-slow", "timeout", "flaky"].includes(reason) ||
-                             (["fail-with-canvas", "needs-canvas"].includes(reason) && !hasCanvas);
-          const expectFail = (reason === "fail") ||
-                             (reason === "fail-with-canvas" && hasCanvas) ||
-                             (reason === "needs-node10" && !hasNode10) ||
-                             (reason === "needs-node11" && !hasNode11) ||
-                             (reason === "needs-node12" && !hasNode12);
-
-          if (shouldSkip) {
-            specify.skip(`[${reason}] ${testFile}`);
-          } else if (expectFail) {
-            const failReason = reason !== "fail" ? `: ${reason}` : "";
-            runSingleWPT(testFilePath, `[expected fail${failReason}] ${testFile}`, expectFail);
-          } else {
-            runSingleWPT(testFilePath, testFile, expectFail);
+          switch (resolveReason(reason)) {
+            case "skip":
+              specify.skip(`[${reason}] ${testFile}`);
+              break;
+            case "expect-fail": {
+              const failReason = reason !== "fail" ? `: ${reason}` : "";
+              runSingleWPT(testFilePath, `[expected fail${failReason}] ${testFile}`, true);
+              break;
+            }
+            default:
+              runSingleWPT(testFilePath, testFile, data || false);
```

Comment:
Can we have a prefix like `[x/y] ` for ones where we expect some failures? If you look at the logs in https://github.com/jsdom/jsdom/pull/3139/checks?check_run_id=2102685649 they read `✓ window-properties.https.html` which is a bit misleading.